### PR TITLE
Remove warnings

### PR DESCRIPTION
**Pull request by pradal**
Lots of warnings :

1. **function prototype** 
warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
2. **'sprintf' is deprecated**
This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
3. **casting void* to int**
warning: cast to 'void *' from smaller integer type 'int' 
## Solutions 
1. Prototypes
```C
MAT	*_m_copy(in,out,i0,j0)
MAT	*in,*out;
u_int	i0,j0;
```
```C
MAT	*_m_copy(MAT* in, MAT* out, u_int i0, u_intj0)
//MAT	*in,*out; 
```
2. sprintf
```C
sprintf(optname,"%s.opt",argv[MIN_OPT]);
```
replaced by
```C
snprintf(optname,sizeof(optname), "%s.opt",argv[MIN_OPT]);
```
3. casting
```C
    shmdt((void*)shmid); //ShMDetach
```
to
```C
    shmdt((void*)(size_t)shmid); //ShMDetach
```

### CODE REVIEW

**Comment by pradal** [comment]
@pellem Here is the instructions

**Comment by pellem** [comment]
Casting:
` fprintf(fp,"ive @ 0x%lx\n",(long)(iv->ive));`
   
Replaced by:
`fprintf(fp,"ive @ 0x%p\n",(size_t)(iv->ive));`

**Comment by pradal** [comment]
@christian34 : We are removing all the warnings... A first step toward refactored caribu

**Review by pradal — COMMENTED** [review]
@pellem 
it seems %p works with void* while size_t works with %ld.
Perhaps you can cast into a void* with %p

**Comment by pellem** [comment]
Replaced _unsigned long_ with _uintptr_t_ for safe pointer conversion:
(Needed to include <stdint.h>

Before:
```
static unsigned int mem_hash(ptr)
void **ptr;
{
   unsigned long lp = (unsigned long)ptr;

   return (lp % MEM_HASHSIZE);
}

```
After:
```
#include <stdint.h> // Ajoute cette ligne en haut du fichier pour utiliser uintptr_t

static unsigned int mem_hash(void **ptr)
{
   uintptr_t lp = (uintptr_t)ptr;

   return (lp % MEM_HASHSIZE);
}
```

**Comment by pellem** [comment]
> @pellem it seems %p works with void* while size_t works with %ld. Perhaps you can cast into a void* with %p

Ok will check it

**Comment by pellem** [comment]
> @pellem it seems %p works with void* while size_t works with %ld. Perhaps you can cast into a void* with %p

Done

**Comment by pradal** [comment]
You can also safely remove **register**

File: src/cpp/Canestra/src/canopy_io.cpp [hunk 1]
```diff
@@ -755,17 +755,17 @@ long int Canopy::read_shm(
   //liberation du shm
 #ifndef WIN32
   // Unix
-  shmdt((void*)shmid);
+  shmdt((void*)(size_t)shmid);
```
Comment (pradal) [inline src/cpp/Canestra/src/canopy_io.cpp:758]: Verify if void* is needed

File: src/cpp/Canestra/src/capteur.cpp [hunk 2]
```diff
@@ -414,8 +414,8 @@ void Camera::calc_visi(Liste <Diffuseur*>& Ldiff)
 void Camera::maj_stat()
  { double x;
 
-   for(register int i=0;i<img->taille(0);i++)
-     for(register int j=0;j<img->taille(1);j++)
+   for(int i=0;i<img->taille(0);i++)
+     for(int j=0;j<img->taille(1);j++)
          { x=img->val(i,j);
```
Comment (pradal) [inline src/cpp/Canestra/src/capteur.cpp:419]: int i=0, j=0;
outside the loop

File: src/cpp/meschach/mesch12a/src/machine.c [hunk 3]
```diff
@@ -99,39 +99,39 @@ register int	len;
 
 /* __smlt__ scalar multiply array c.f. sv_mlt() */
 void	__smlt__(dp,s,out,len)
-register Real	*dp, *out;
-register double s;
-register int	len;
+ Real	*dp, *out;
+ double s;
+ int	len;
```
Comment (pradal) [inline src/cpp/meschach/mesch12a/src/machine.c:104]: prototype : add the parameters in the function def

File: src/cpp/meschach/mesch12a/src/machine.c [hunk 4]
```diff
@@ -99,39 +99,39 @@ register int	len;
...
     for ( i = 0; i < len; i++ )
 	out[i] = s*dp[i];
 }
 
 /* __add__ -- add arrays c.f. v_add() */
 void	__add__(dp1,dp2,out,len)
-register Real	*dp1, *dp2, *out;
-register int	len;
+ Real	*dp1, *dp2, *out;
+ int	len;
```
Comment (pradal) [inline src/cpp/meschach/mesch12a/src/machine.c:114]: idem

File: src/cpp/meschach/mesch12a/src/machine.c [hunk 5]
```diff
@@ -99,39 +99,39 @@ register int	len;
...
     for ( i = 0; i < len; i++ )
 	out[i] = dp1[i] + dp2[i];
 }
 
 /* __sub__ -- subtract arrays c.f. v_sub() */
 void	__sub__(dp1,dp2,out,len)
-register Real	*dp1, *dp2, *out;
-register int	len;
+ Real	*dp1, *dp2, *out;
+ int	len;
```
Comment (pradal) [inline src/cpp/meschach/mesch12a/src/machine.c:124]: idem

Comment (pellem) [reply src/cpp/meschach/mesch12a/src/machine.c]: done

File: src/cpp/meschach/mesch12a/src/machine.c (hunk 4 above)
Comment (pellem) [reply src/cpp/meschach/mesch12a/src/machine.c]: done

File: src/cpp/meschach/mesch12a/src/machine.c (hunk 3 above)
Comment (pellem) [reply src/cpp/meschach/mesch12a/src/machine.c]: done

File: src/cpp/Canestra/src/capteur.cpp (hunk 2 above)
Comment (pellem) [reply src/cpp/Canestra/src/capteur.cpp:419]: What is the most effective way to do that ?

Comment (pradal) [reply src/cpp/Canestra/src/capteur.cpp:419]: Just do 
```C
int i=0, j=0;
for(i=0;i<img->taille(0);i++)
     for( j=0;j<img->taille(1);j++)
         { x=img->val(i,j);
```

**Comment by pellem** [comment]
@pradal Can you check the updated code before we merge the PR please ? :)

**Comment by pradal** [comment]
Congrats! Everything needed is fixed!
Close issue #56

**Pull request merged**